test/e2e/pkg/consumer_test.go

```diff
@@ -0,0 +1,72 @@
+package e2e_test
+
+import (
+	"net/http"
+	"reflect"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-online/maestro/pkg/api/openapi"
+)
+
+// go test -v ./test/e2e/pkg -args -api-server=$api_server -consumer-name=$consumer_name -consumer-kubeconfig=$consumer_kubeconfig -ginkgo.focus "Server Side API"
+var _ = Describe("Consumer", Ordered, func() {
+	var testConsumerName string
+	var testConsumerID string
+	BeforeAll(func() {
+		testConsumerName = "test-consumer"
+	})
+
+	Context("Consumer CRUD Tests", func() {
+		It("list the default consumer", func() {
+			consumerList, resp, err := apiClient.DefaultApi.ApiMaestroV1ConsumersGet(ctx).Execute()
+			Expect(err).To(Succeed())
+			Expect(resp.StatusCode).To(Equal(http.StatusOK))
+			Expect(consumerList).NotTo(BeNil())
+			Expect(len(consumerList.Items) > 0).To(BeTrue())
+
+			got := false
+			for _, consumer := range consumerList.Items {
+				if *consumer.Name == consumerOpts.Name {
+					got = true
+				}
+			}
+			Expect(got).To(BeTrue())
```

Comment:
how to guarantee the consumer is there? the test order is random. if run this test at the beginning, the consumer should be example. right?

Reply:
For this case, we already have generated a consumer before running the e2e test. So the consumer must be there~

Reply:
How about the case you want to run it in an existing environment? Do you require create the consumer as a pre-requirement? maybe just adjust your order to create/get/patch